sort.h

```diff
@@ -877,20 +877,21 @@ static void TIM_SORT_MERGE(SORT_TYPE *dst, const TIM_SORT_RUN_T *stack, const in
   } else {
     /* right merge */
     memcpy(storage, &dst[curr + A], B * sizeof(SORT_TYPE));
-    i = B - 1;
-    j = curr + A - 1;
+    i = B;
+    j = curr + A;
+    k = curr + A + B;
 
-    for (k = curr + A + B - 1; k >= curr; k--) {
-      if ((i >= 0) && (j >= curr)) {
-        if (SORT_CMP(dst[j], storage[i]) > 0) {
-          dst[k] = dst[j--];
+    while (k-- > curr) {
```

Comment:
I'm curious why we switched this loop from being indexed by `i` and `j` to `i - 1` and `j - 1`. Is there a performance impact?

Reply:
With unsigned indices, the condition like `i >= 0` is always true. The same with `j >= curr` and `k >= curr` if `curr` is zero.